lib/internal/util.js

```diff
@@ -275,24 +272,25 @@ function promisify(original) {
   const argumentNames = original[kCustomPromisifyArgsSymbol];
 
   function fn(...args) {
-    const promise = createPromise();
-    try {
-      original.call(this, ...args, (err, ...values) => {
-        if (err) {
-          promiseReject(promise, err);
-        } else if (argumentNames !== undefined && values.length > 1) {
-          const obj = {};
-          for (var i = 0; i < argumentNames.length; i++)
-            obj[argumentNames[i]] = values[i];
-          promiseResolve(promise, obj);
-        } else {
-          promiseResolve(promise, values[0]);
-        }
-      });
-    } catch (err) {
-      promiseReject(promise, err);
-    }
-    return promise;
+    return new Promise((resolve, reject) => {
```

Comment:
In fact, the only reason we even have promisify in core is to do the above optimization and avoid the closure - as well as enable further optimizations. We definitely need to run the benchmarks on this one IMO.

Reply:
> In fact, the only reason we even have promisify in core is to do the above optimization

I do not agree with that. There are likely different opinions about this.

> as well as enable further optimizations

This is actually prevented with the current approach.